pkg/action/executor/move_util.go

```diff
@@ -281,6 +280,24 @@ func deleteInvalidPendingPods(parent *unstructured.Unstructured, podClient v1.Po
 
 func newPendingInvalidPods(list1, list2 *api.PodList) []*api.Pod {
 	var diffList []*api.Pod
+	if list2 == nil {
```

Comment:
@irfanurrehman, sorry I wasn't able to talk to you last night, I would like to catch up with you maybe tonight to understand under what circumstances `list1` or `list2` would become nil?

But just looking at this function overall, I was wondering if it can be simplified to the following?

```
func newPendingInvalidPods(list1, list2 *api.PodList) (diffList []*api.Pod) {
	if list1 == nil {
		return
	}
	for _, pod1 := range list1.Items {
		found := false
		if list2 != nil {
			for _, pod2 := range list2.Items {
				if pod1.Name == pod2.Name {
					// pod1 from list1 also exists in list2
					found = true
				}
			}
		}
		if found == true {
			continue
		}
		if pod1.Status.Phase == api.PodPending &&
			pod1.Spec.SchedulerName == "turbo-scheduler" {
			diffList = append(diffList, &pod1)
		}
	}
	return
}
```

Reply:
@ading1977 thanks for the suggestion. Updated!